src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/Tool/AppProvisioningTool.cs

```diff
@@ -231,12 +231,13 @@ public AppProvisioningTool(string commandName, ProvisioningToolOptions provision
                         ConsoleLogger.LogMessage(failMessage, LogMessageType.Error);
                     }
                 }
-                catch (Exception ex)
+                catch (ServiceException ex)
                 {
+                    string jsonErrorContent = $"{ex.Error?.Code ?? string.Empty} {Environment.NewLine} {ex.Error?.Message ?? string.Empty}";
                     JsonResponse jsonResponse = new JsonResponse(CommandName)
                     {
                         State = State.Fail,
```

Comment:
Why not JsonResponse has a State, Content, ErrorCode set of properties?  The Content of an Fail could be the error message quite reasonably, and ErrorCode only set during Fail.